.github/workflows/main.yml

```diff
@@ -71,22 +71,43 @@ jobs:
           echo "Highest released version was: $REPO_VERSION_MOST_RECENT, desired version (specified in Cargo.toml) is $REPO_VERSION_DESIRED"
 
           RELEASE="false"
+          PRERELEASE="false"
           if [[ "$IS_FORK" == "false" ]]
           then
-            if [[ "$BRANCH_NAME" == "master" && "$REPO_VERSION_DESIRED" != "$REPO_VERSION_MOST_RECENT" ]]
+            if [[ "$REPO_VERSION_DESIRED" != "$REPO_VERSION_MOST_RECENT" ]]
             then
-              PUBLISH_VERSION="$REPO_VERSION_DESIRED"
-              RELEASE="true"
-              echo "This is push to master, and version was bumped $REPO_VERSION_MOST_RECENT->$REPO_VERSION_DESIRED. Will publish and release $REPO_VERSION_DESIRED."
+              if [[ "$BRANCH_NAME" == "master" ]]
+              then
+                PUBLISH_VERSION="$REPO_VERSION_DESIRED"
+                RELEASE="true"
+                echo "This is push to master, and version was bumped from $REPO_VERSION_MOST_RECENT to $REPO_VERSION_DESIRED. Will publish a release of $REPO_VERSION_DESIRED."
+              else
+                PUBLISH_VERSION="$REPO_VERSION_DESIRED-${{ github.run_number }}"
+                PRERELEASE="true" 
+                echo "This is push to a non-master branch, and version was bumped from $REPO_VERSION_MOST_RECENT to $REPO_VERSION_DESIRED. Will publish a pre-release of $REPO_VERSION_DESIRED."
+              fi;
             else
               PUBLISH_VERSION="$REPO_VERSION_MOST_RECENT-$BRANCH_NAME-${{ github.run_number }}"
-              echo "This is not push to master. Will be publishing version: $PUBLISH_VERSION."
+              if [[ ${{ contains(github.event.pull_request.labels.*.name, 'ci') }} ]]
```

Comment:
Will this 
```
              if [[ ${{ contains(github.event.pull_request.labels.*.name, 'ci') }} ]]
```
also work if I am running CI on a branch, not PR? I think if it's branch, field `pull_request` would not exist, so it might be broken in that case. 
Sometimes I run pipelines on branches

Reply:
I think the solution might be to just not allow making pre-releases from branches.